### DISCUSSION TITLE
Xiaomi disconnections probably fixed in Routers

### DISCUSSION THREAD

**Discussion by FlyingDomotic** [Koenkk/zigbee2mqtt#13675] [General]
I probably find a way to fix Xiaomi disconnection issues (I've more than 80 Lumi/Xiaomi devices, so I'm a bit concerned ;-)

As of now, I only tested with CC2652B, but on the principle, it should work with all TI devices.

I didn't get any disconnection for more than a week, while it was common to loose 30/40 devices per day...

I just reproduced changes done in coordinator about child again and parent announce.

In details, I applied a change in Stack/sys/zglobals.c , setting
```
uint8_t zgChildAgingEnable = FALSE;" 
```
and changed twice the paragraph:
```
       if ( zgChildAgingEnable == TRUE )
        {
          // Once the device has changed its state to a ROUTER set the timer to send
          // Parent annce
          ZDApp_SetParentAnnceTimer();
        }

```
by :
```
          ZDApp_SetParentAnnceTimer();
```
in Stack/zdo/zd_app.c

I didn't reproduce manufacturer code change, as it seems only useful in Coordinator. I'm also not so sure that parent announce modification is needed ZDApp_NetworkStartEvt, as I still miss detailed understanding of zStack, but in worst case, it's just useless.

Hope this will help!

**Comment by Koenkk** [comment]
Thanks, I will integrate these changes the next time I update the router fw!

**Reply by ut1545** [reply]
Koenkk,
Could you please tell us when you plan to release the new router firmware? 
Thanks in advance,

**Reply by Koenkk** [reply]
I'm currently doing some final checks with the coordinator version, once that is ready I will also update the router.

**Comment by Koenkk** [comment]
Router fw with this change is available now: https://github.com/Koenkk/Z-Stack-firmware/tree/develop/router/Z-Stack_3.x.0/bin

**Comment by jerrm** [comment]
Should the version update?

Using the HA addon...

I flashed:
![image](https://user-images.githubusercontent.com/12073505/189175665-d4d7388e-6a0a-44ea-9bfe-b4e354e1d5cf.png)

Then re-paired without removing from z2m.  Version still showed 20220125:
![image](https://user-images.githubusercontent.com/12073505/189173161-9a0a9d8b-dd8c-44a1-9482-054766f33910.png)

Version number persisted after restarting the add on.

I then removed and re-paired with the same results:
![image](https://user-images.githubusercontent.com/12073505/189175967-b237d039-7062-4783-8670-ff830adc3afb.png)

**Reply by Koenkk** [reply]
Force remove, restart z2m and pair again.

**Reply by jerrm** [reply]
Thanks, that did it.

I had to re-flash to get the stick back into pairing mode.  Next time I'll know to force remove before the first flash.

Any chance we could see a router backup/restore to avoid re-pairing?

**Reply by Koenkk** [reply]
> Any chance we could see a router backup/restore to avoid re-pairing?

OTA maybe in the future

**Comment by ut1545** [comment]
Koenkk,
I have updated my router a while and all xiaomi devices are working perfectly. Thank you very much,
